src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultTagHelperDescriptorBuilder.cs

```diff
@@ -135,47 +134,13 @@ public override void TagMatchingRule(Action<TagMatchingRuleDescriptorBuilder> co
 
     public override TagHelperDescriptor Build()
     {
-        var diagnostics = new HashSet<RazorDiagnostic>();
-        if (_diagnostics != null)
-        {
-            diagnostics.UnionWith(_diagnostics);
-        }
-
-        var allowedChildTags = Array.Empty<AllowedChildTagDescriptor>();
-        if (_allowedChildTags != null)
-        {
-            var allowedChildTagsSet = new HashSet<AllowedChildTagDescriptor>(AllowedChildTagDescriptorComparer.Default);
-            for (var i = 0; i < _allowedChildTags.Count; i++)
-            {
-                allowedChildTagsSet.Add(_allowedChildTags[i].Build());
-            }
+        using var diagnostics = new PooledHashSet<RazorDiagnostic>();
```

Comment:
Why not `using` these? It's a ref struct, so pattern-based dispose should work.

Reply:
Because they're passed by ref, so the using pattern doesn't work:

![image](https://user-images.githubusercontent.com/116161/223285423-9f01aec6-a832-423f-be28-bdbcfb38d0f3.png)

Reply:
Gotcha, thanks.